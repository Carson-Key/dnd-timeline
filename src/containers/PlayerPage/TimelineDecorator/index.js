import React, { Component } from 'react'
import {
  Button,
  Container,
  Jumbotron
} from 'reactstrap';
import Time from '../../Time'
import { capitalizeFirstLetter } from '../../../utilities/capitalizeFirstLetter.js'
import { addPossesive } from '../../../utilities/addPossesive.js'

class TimelineDecorator extends Component {
  render() {
    return (
      <div>
        <Button
          className="exitButton"
          color="danger"
          onClick={this.props.exitTimeLine}
        >
          exit {capitalizeFirstLetter(this.props.characterName)}
        </Button>
        <Jumbotron fluid>
          <Container fluid>
            <center>
              <h1 className="display-3">
                {addPossesive(capitalizeFirstLetter(this.props.characterName))} Timeline
              </h1>
            </center>
          </Container>
        </Jumbotron>
        <Time
          timeline={this.props.playerObject.characters[this.props.characterName].timeline}
          lessThanYearZero={this.props.lessThanYearZero}
          greaterThanYearZero={this.props.greaterThanYearZero}
        />
      </div>
    )
  }
}

export default TimelineDecorator
