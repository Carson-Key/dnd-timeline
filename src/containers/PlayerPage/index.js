import React, { Component } from 'react'
import './index.css'
import CharacterEntry from '../CharacterEntry'
import Timeline from '../Timeline'

class PlayerPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      characterName: "",
      moveOn: ""
    }

    this.exitTimeLine = this.exitTimeLine.bind(this);
    this.characterSelect = this.characterSelect.bind(this);
  }

  characterSelect(name) {
    this.setState({
      characterName: name,
      moveOn: "go"
    })
  }

  exitTimeLine() {
    this.setState({
      moveOn: ""
    });
  }

  render() {
    if (this.state.moveOn === "") {
      return (
        <CharacterEntry
          playerObject={this.props.playerObject}
          characterSelect={this.characterSelect}
        />
      )
    } else {
      return (
        <Timeline
          exitTimeLine={this.exitTimeLine}
          characterName={this.state.characterName}
          playerObject={this.props.playerObject}
          lessThanYearZero={this.props.lessThanYearZero}
          greaterThanYearZero={this.props.greaterThanYearZero}
        />
      )
    }
  }
}

export default PlayerPage
