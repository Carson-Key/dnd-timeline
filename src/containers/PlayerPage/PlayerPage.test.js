import React from "react"
import PlayerPage from "./index.js"
import { shallow, render } from 'enzyme'
import toJson from "enzyme-to-json";

describe('Render PlayerPage ->', () => {

  test('with minimal props', () => {
    const playerPage = render(
      <PlayerPage
        playerObject={
          {
            timelines: {
              ameria: {
                name: "Ameria",
                characters: {
                  "cade": {
                      "name": "Cade Goodbarrel",
                      "characterKey": "cade",
                      "timelineKey": "ameria",
                      "timeline": {
                        "events": {}
                      }
                    }
                }
              }
            }
          }
        }
       />,
    )

    expect(toJson(playerPage)).toMatchSnapshot()
  })

  test('after setting state moveOn to go, page should render Timeline', () => {
    const playerPage = shallow(
      <PlayerPage
        playerObject={
          {
            timelines: {
              ameria: {
                name: "Ameria",
                characters: {
                  "cade": {
                      "name": "Cade Goodbarrel",
                      "characterKey": "cade",
                      "timelineKey": "ameria",
                      "timeline": {
                        "events": {}
                      }
                    }
                }
              }
            }
          }
        }
       />,
    )

    playerPage.setState({moveOn: "go", characterName: "cade"})

    expect(toJson(playerPage)).toMatchSnapshot();
  })

})
