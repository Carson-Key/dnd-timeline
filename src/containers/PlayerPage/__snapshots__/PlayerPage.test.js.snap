// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PlayerPage -> after setting state moveOn to go, page should render Timeline 1`] = `
<Timeline
  characterName="cade"
  exitTimeLine={[Function]}
  playerObject={
    Object {
      "timelines": Object {
        "ameria": Object {
          "characters": Object {
            "cade": Object {
              "characterKey": "cade",
              "name": "Cade Goodbarrel",
              "timeline": Object {
                "events": Object {},
              },
              "timelineKey": "ameria",
            },
          },
          "name": "Ameria",
        },
      },
    }
  }
  timelineName=""
/>
`;

exports[`Render PlayerPage -> with minimal props 1`] = `
<div
  class="container"
>
  <br />
  <div
    class="row"
  >
    <div
      class="col"
    >
      <center>
        <p>
          Pick a character
        </p>
        <br />
      </center>
      <center>
        <p>
          Ameria
        </p>
        <br />
      </center>
      <center>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Cade Goodbarrel
        </button>
        <br />
        <br />
      </center>
    </div>
  </div>
</div>
`;
